Add migration steps for conda-store namespaces (#178)
nebari-dev/nebari-docs: docs/docs/troubleshooting.mdx

## Configuration

### Conda-store compatibility migration steps when upgrading to 0.4.5

If you are upgrading from a version of Nebari prior to `0.4.5`, you will need to manually update your conda-store namespaces
to be compatible with the new Nebari version. This is a one-time migration step that will need to be performed after upgrading to continue using the service.

> After upgrading to `0.4.5`, the base names for the follow namespaces that are built in with Nebari were changed as follows:

Reviewer: ```suggestion
After upgrading to `0.4.5`, the base names for the following namespaces that are built in with Nebari were changed as follows:
```

Second, you will need to create a new JupyterLab profile to select the GPU node.
In this way, you'll have a separate profile for GPUs which users can select.

### Migrate existing environments from a namespaces to another

> When the user creates a new environment, it requires selecting a namespace. This namespace is used to identify the group permissions to

Reviewer: ```suggestion
When you create a new environment, it requires selecting a namespace. This namespace is used to identify the group permissions to
```

that environment and the location where the such environment needs to be stored.

> To migrate environments from one namespace to another, the user must manually build all environments in the new namespace.

Reviewer: ```suggestion
To migrate environments from one namespace to another, you must manually build all environments in the new namespace.
```

Reviewer: @viniciusdc A quick question so that I understand this better, do just the administrators (i.e., people managing the qhub/nebari deployment and user groups) need to do this migration, or do all the people (for example, I'm thinking data scientists using JupyterLab) using the platform?

Author: That depends a lot on the excatly situation:
- If the user (data scientist) has permission over the namespace (e.g. its currently assigned group matches the target namespace), then its possible for them to do these changes
- Admins normally have access to all namespaces, so this extra validation is not needed
- There is only the caveat of the Nebari system managed namespaces `nebari-git` (currently `filesystem`), and `global` (`default`), wich normally are completely restricted to anyone -- to change this a Cluster administrator needs to intervene

Reviewer: Got it, thanks!

> Once migrated, the user can delete the old namespace, permanently deleting all existing environments built against it.

Reviewer: ```suggestion
Once migrated, you can delete the old namespace, permanently deleting all existing environments built against it.
```

> Once migrated, the user can delete the old namespace, permanently deleting all existing environments built against it.

Author: unfortunate not, I added the endpoint link to the namespaces UI page as an attempt to showcase the `delete` button in there, but I agree that an in-depth section somewhere would be good -- I think this ties in with the conda-store management docs will be written

### Handle Access restricted namespaces

> The built in namespace (`nebari-git`) is used to store the environments contained in the nebari-config.yaml file.

Author: `nebari-git` corresponds to `filesystem`, so unless you have special permissions you can't delete/update/add any environments in there :(

> The built in namespace (`nebari-git`) is used to store the environments contained in the nebari-config.yaml file.

Reviewer 2: You're right, sorry for the noise

As this namespace is controlled by the Nebari conda-store server and not the user, it is not possible to modify
or delete it without proper permissions.

> To modify this namespace, the user need to manually update the `conda-store-config.py` configmap located in the

Reviewer: Similar question to earlier, do the administrators need to do this or individual users?

Second, you will need to create a new JupyterLab profile to select the GPU node.
In this way, you'll have a separate profile for GPUs which users can select.

### Migrate existing environments from a namespaces to another

When you create a new environment as a user, it requires selecting a namespace. This namespace is used to identify the group permissions to

> that environment and the location where the such environment needs to be stored.

Reviewer: ```suggestion
that environment and the location where the environment needs to be stored.
```

To migrate environments from one namespace to another, you must manually build all environments in the new namespace.
Once migrated, you can delete the old namespace, permanently deleting all existing environments built against it.

### Handle Access to restricted namespaces

> The built in namespace `nebari-git` is used to build the environments from the nebari-config.yaml file.

Reviewer: ```suggestion
The built in namespace `nebari-git` is used to build the environments from the `nebari-config.yaml` file.
```
